python/lsst/meas/extensions/scarlet/scarletDeblendTask.py

```diff
@@ -328,6 +331,136 @@ def deblend(mExposure, footprint, config):
+
+    # Use the inverse variance as the weights
+    if config.useWeights:
+        weights = 1/mExposure.variance[:, bbox].array
+    else:
+        weights = np.ones_like(images)
+        badPixels = mExposure.mask.getPlaneBitMask(config.badMask)
+        mask = mExposure.mask[:, bbox].array & badPixels
+        weights[mask > 0] = 0
+
+    # Mask out the pixels outside the footprint
+    mask = getFootprintMask(footprint, mExposure)
+    weights *= ~mask
+
+    psfs = _computePsfImage(mExposure, footprint.getCentroid()).array.astype(np.float32)
+    modelPsf = lite.integrated_circular_gaussian(sigma=config.modelPsfSigma)
+
+    observation = lite.LiteObservation(
+        images=images,
+        variance=variance,
+        weights=weights,
+        psfs=psfs,
+        model_psf=modelPsf[None, :, :],
+        convolution_mode=config.convolutionType,
+    )
+
+    # Convert the centers to pixel coordinates
+    xmin = bbox.getMinX()
+    ymin = bbox.getMinY()
+    centers = [
+        np.array([peak.getIy() - ymin, peak.getIx() - xmin], dtype=int)
+        for peak in footprint.peaks
+        if not isPseudoSource(peak, config.pseudoColumns)
+    ]
+
+    # Initialize the sources
+    if config.morphImage == "chi2":
+        sources = lite.init_all_sources_main(
+            observation,
+            centers,
+            min_snr=config.minSNR,
+            thresh=config.morphThresh,
+        )
+    elif config.morphImage == "wavelet":
+        _bbox = bboxToScarletBox(len(mExposure.filters), bbox, bbox.getMin())
+        _wavelets = wavelets[(slice(None), *_bbox[1:].slices)]
+        sources = lite.init_all_sources_wavelets(
+            observation,
+            centers,
+            use_psf=False,
+            wavelets=_wavelets,
+            min_snr=config.minSNR,
+        )
+    else:
+        raise ValueError("morphImage must be either 'chi2' or 'wavelet'.")
+
+    # Set the optimizer
+    if config.optimizer == "adaprox":
+        parameterization = partial(
+            lite.init_adaprox_component,
+            bg_thresh=config.backgroundThresh,
+            max_prox_iter=config.maxProxIter,
+        )
+    elif config.optimizer == "fista":
+        parameterization = partial(
+            lite.init_fista_component,
+            bg_thresh=config.backgroundThresh,
+        )
+    sources = lite.parameterize_sources(sources, observation, parameterization)
+
+    # Attach the peak to all of the initialized sources
+    for k, center in enumerate(centers):
+        # This is just to make sure that there isn't a coding bug
+        if len(sources[k].components) > 0 and np.any(sources[k].center != center):
+            raise ValueError("Misaligned center, expected {center} but got {sources[k].center}")
+        # Store the record for the peak with the appropriate source
+        sources[k].detectedPeak = footprint.peaks[k]
+
+    blend = lite.LiteBlend(sources, observation)
+
+    # Initialize each source with its best fit spectrum
```

Comment:
It's a least squares fit that uses the morphology of each source to simultaneously fit their initial spectra. This has a huge impact on the number of iterations and overall final logL. But because it requires all of the sources to be projected on the full box of the blend, for large blends and/or dense regions there is a memory problem fitting them simultaneously.

Reply:
Ok, so you do a linear least squares fit for fluxes first, yes? If it's memory-constrained, can you use sparse matrices?